File: client/src/types/apiTypes.ts
export interface UploadFileResponse {
  chatId: string;
  message: string;
}

export interface DocumentResponse {
  id: string;
  page: number;
  content: string;
}

export interface QueryResponse {
  question: string;
  answer: string;
  documents: DocumentResponse[];
}

export interface ChatInfo {
  chatId: string;
  fileName: string;
  createdAt: string;
}

export interface ChatsResponse {
  userId: string;
  chats: ChatInfo[];
}

export interface Message {
  role: string;
  content: string;
  timestamp: string;
  documents: DocumentResponse[];
}

export interface ChatHistoryResponse {
  chatId: string;
  history: Message[];
}

File: api/src/doc_chat/api_types.py
from pydantic import BaseModel


class ChatQueryRequest(BaseModel):
    chatId: str
    question: str


class UploadFileResponse(BaseModel):
    chatId: str
    message: str


class Message(BaseModel):
    role: str
    content: str
    timestamp: str


class ChatHistoryResponse(BaseModel):
    chatId: str
    history: list[Message]


class Chat(BaseModel):
    chatId: str
    fileName: str
    createdAt: str


class ChatsResponse(BaseModel):
    userId: str
    chats: list[Chat]


class DocumentsResponse(BaseModel):
    id: str
    page: int
    content: str


class QueryResponse(BaseModel):
    question: str
    answer: str
    documents: list[DocumentsResponse]
